pkg/scaffold/olm-catalog/csv_test.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2018 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package catalog
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/operator-framework/operator-sdk/internal/util/diffutil"
+	"github.com/operator-framework/operator-sdk/pkg/scaffold"
+	"github.com/operator-framework/operator-sdk/pkg/scaffold/input"
+
+	"github.com/coreos/go-semver/semver"
+	"github.com/ghodss/yaml"
+	olmApi "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	olmInstall "github.com/operator-framework/operator-lifecycle-manager/pkg/controller/install"
+)
+
+const testDataDir = "testdata"
+
+func TestCSV(t *testing.T) {
+	buf := &bytes.Buffer{}
+	s := &scaffold.Scaffold{
+		GetWriter: func(_ string, _ os.FileMode) (io.Writer, error) {
+			return buf, nil
+		},
+	}
+	csvVer := "0.1.0"
+	projectName := "app-operator"
+	testOLMDir := filepath.Join(testDataDir, scaffold.OLMCatalogDir)
+
+	err := s.Execute(&input.Config{ProjectName: projectName},
+		&CSV{
+			CSVVersion:     csvVer,
+			ConfigFilePath: filepath.Join(testOLMDir, CSVConfigYamlFile),
+		},
+	)
+	if err != nil {
+		t.Fatalf("Failed to execute the scaffold: (%v)", err)
+	}
+
+	// Get the expected CSV manifest from test data dir.
+	csvFilePath := projectName + CSVYamlFileExt
+	csvExpBytes, err := ioutil.ReadFile(filepath.Join(testOLMDir, csvFilePath))
+	if err != nil {
+		t.Fatal(err)
+	}
+	csvExp := string(csvExpBytes)
+
+	if csvExp != buf.String() {
+		diffs := diffutil.Diff(csvExp, buf.String())
+		t.Fatalf("Expected vs actual differs.\n%v", diffs)
+	}
+}
+
+func TestUpdateVersion(t *testing.T) {
+	projectName := "app-operator"
+	testOLMDir := filepath.Join(testDataDir, scaffold.OLMCatalogDir)
+	csvFilePath := projectName + CSVYamlFileExt
+	csvExpBytes, err := ioutil.ReadFile(filepath.Join(testOLMDir, csvFilePath))
+	if err != nil {
+		t.Fatal(err)
+	}
+	csv := &olmApi.ClusterServiceVersion{}
+	if err := yaml.Unmarshal(csvExpBytes, csv); err != nil {
+		t.Fatal(err)
+	}
+
+	newCSVVer := "0.2.0"
+	c := &CSV{
+		Input: input.Input{
+			ProjectName: projectName,
+		},
+		CSVVersion: newCSVVer,
+	}
+	if err := c.updateCSVVersions(csv); err != nil {
+		t.Fatalf("Update csv with ver %s: (%v)", newCSVVer, err)
```

Comment:
```suggestion
		t.Fatalf("Failed to update csv with version %s: (%v)", newCSVVer, err)
```
It's best if we keep the test error messages as clear as possible on what failed.